[Build] Change build to webpack

VWoeltjen opened the pull request:
Fixes #4; instead of using RequireJS for loading/optimizing, switch to webpack, which has cleaner support for SCSS dependencies. Has a nice side effect of cleaning out a lot of unused build code.

### Author Checklist

1. Changes address original issue? Y
2. Unit tests included and/or updated with changes? N (no test suite yet; filed #5)
3. Command line build passes? Y
4. Changes have been smoke-tested? Y

VWoeltjen committed:
[Build] Initial webpack build

Begin transitioning to webpack to simplify SASS build.
Fixes #4

VWoeltjen committed:
[Build] Handle ES6-style modules from Babel

VWoeltjen committed:
[Build] Use vue with included compiler

VWoeltjen committed:
[Build] Update dev environment instructions

VWoeltjen committed:
[Build] Enable source maps

VWoeltjen committed:
[Build] Minify

VWoeltjen committed:
[Build] Run webpack during prepublish step

VWoeltjen committed:
[Build] Change heading depth in README

VWoeltjen committed:
[Build] Remove RequireJS usage; just load UMD scripts

VWoeltjen committed:
[Build] Decruft after webpack conversion

VWoeltjen committed:
[Build] Remove unused dependencies

VWoeltjen committed:
[Build] Upgrade UglifyJsPlugin

https://github.com/webpack/webpack-dev-server#caveats

VWoeltjen committed:
[Build] Reorganize sources

VWoeltjen committed:
[Build] Enable source maps, again

VWoeltjen committed:
[Build] Switch from prepublish to prepare

...to simplify usage as a repository URL dependency

VWoeltjen committed:
[Build] Include dist as files in package.json

akhenry commented on webpack.config.js:
```diff
@@ -0,0 +1,54 @@
+var webpack = require('webpack');
```
Unused import?

akhenry reviewed (dismissed):
This looks good to me (aside from the possible unused import?)

Have not worked with webpack before, but it looks like it could streamline a lot of the build process and replace a bunch of our build dependencies and scripting in master if we were to roll it out there.

VWoeltjen committed:
[Build] Remove unused import

https://github.com/VWoeltjen/openmct-heatmap/pull/6/files/a32b2d65eec9fb026404554b8599eb035d6587e9#r160503461

VWoeltjen commented:
Addressed issue from review. Foregoing re-review or review checklist, as this is still a demo component
